exporter/awsemfexporter/datapoint.go

```diff
@@ -122,13 +134,14 @@ func (dps numberDataPointSlice) At(i int) (dataPoint, bool) {
 		metricVal = float64(metric.IntValue())
 	}
 
-	retained := true
+	var retained bool = true
+	var datapoints []dataPoint = []dataPoint{}
+
 	if dps.adjustToDelta {
 		var deltaVal interface{}
-		deltaVal, retained = deltaMetricCalculator.Calculate(dps.metricName, mergeLabels(dps.deltaMetricMetadata, labels),
-			metricVal, metric.Timestamp().AsTime())
+		deltaVal, retained = deltaMetricCalculator.Calculate(dps.metricName, mergeLabels(dps.deltaMetricMetadata, labels), metricVal, metric.Timestamp().AsTime())
 		if !retained {
-			return dataPoint{}, retained
+			return datapoints, retained
```

Comment:
```suggestion
			return nil, retained
```

This doesn't even need to be an allocated slice since `nil` and an empty slice can be used interchangeably in most situations.

Reply:
I'm fine with that. The reason why I giving its a variable is to keep consistency between calculateDeltaDatapoints function.